operator/cmd/flags.go

```diff
@@ -290,13 +290,17 @@ func init() {
 	option.BindEnv(Vp, option.EnableCiliumEndpointSlice)
 
 	flags.Int(operatorOption.CESMaxCEPsInCES, operatorOption.CESMaxCEPsInCESDefault, "Maximum number of CiliumEndpoints allowed in a CES")
-	flags.MarkHidden(operatorOption.CESMaxCEPsInCES)
 	option.BindEnv(Vp, operatorOption.CESMaxCEPsInCES)
 
 	flags.String(operatorOption.CESSlicingMode, operatorOption.CESSlicingModeDefault, "Slicing mode define how ceps are grouped into a CES")
-	flags.MarkHidden(operatorOption.CESSlicingMode)
```

Comment:
Do you remember why those were marked hidden? For flags that are present in both the agent and the operator, we often make the operator ones hidden because we don't expect users to use them directly. It will automatically get passed to the operator if they rely on the ConfigMap.

For that reason, I'd expect `EnableCiliumEndpointSlice` above to also be marked hidden.

Reply:
I assume they were hidden because they aren’t used by default -- `EnableCiliumEndpointSlice` is false by default.
I see that might be a good reason to hide them.

However, I find it useful to see these flags on operator startup.
It can help with debugging and troubleshooting to see all flags and their values on startup.

Do you think it makes sense to get the value of `EnableCiliumEndpointSlice` and to hide the following 4 flags only if it's `false`?

Reply:
> I assume they were hidden because they aren’t used by default -- `EnableCiliumEndpointSlice` is false by default.
> I see that might be a good reason to hide them.

We don't usually mark flags hidden because they are unused by default.

> It can help with debugging and troubleshooting to see all flags and their values on startup.

Aren't all those flags already present in agent logs?

> Do you think it makes sense to get the value of EnableCiliumEndpointSlice and to hide the following 4 flags only if it's false?

I think it would be worth bringing that discussion on #development to get the opinion of others. Seems fine to me, although we might want to do that for other features as well.

I don't think this is blocking for this PR in any case.

Reply:
Only `EnableCiliumEndpointSlice` flag is present in agent logs, but the other 4 flags aren't, because they are specific to operators' logic on how to batch CiliumEndpoints into slices.
Agent doesn't care about it.

Even if `EnableCiliumEndpointSlice` flag exists for both operator and agent, there are cases when we wouldn't want the flags to match.
For example, when we are migrating to CEP batching (to use slices), we might want to enable the flag in the operator first, so it generates all the slices, before agents start watching them (still uses CEPs). And only activate it later for agents.

Thank you for the review. I will bring this up in the development channel on slack.

Reply:
> Only the `EnableCiliumEndpointSlice` flag is present in agent logs, but the other 4 flags aren't, because they are specific to operators' logic on how to batch CiliumEndpoints into slices.

Ah, ok. Then yeah, they probably shouldn't be hidden.

Reply:
Just to provide some context I think these ones were marked hidden in the first place because users who are not familiar with how CES works can hardly tune these values to their benefits and a bad value in these arguments could easily break the cluster... but as we add more tests and CES is becoming more battled, I'm not opposed to open these :)